velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -130,45 +224,80 @@ class JsonParseFunction : public exec::VectorFunction {
   }
 
  private:
-  simdjson::error_code parse(size_t size) const {
+  simdjson::error_code parse(size_t size, JsonViewPtr& jsonView) const {
     simdjson::padded_string_view paddedInput(
         paddedInput_.data(), size, paddedInput_.size());
     SIMDJSON_ASSIGN_OR_RAISE(auto doc, simdjsonParse(paddedInput));
-    SIMDJSON_TRY(validate<simdjson::ondemand::document&>(doc));
+    SIMDJSON_TRY(validate<simdjson::ondemand::document&>(doc, jsonView));
     if (!doc.at_end()) {
       return simdjson::TRAILING_CONTENT;
     }
     return simdjson::SUCCESS;
   }
 
   template <typename T>
-  static simdjson::error_code validate(T value) {
+  static simdjson::error_code validate(T value, JsonViewPtr& jsonView) {
```

Comment:
It looks to me we're spreading the canonicalization changes to `JsonView::canonicalize()` and `JsonParseFunction::validate()`, with `JsonParseFunction::validate()` doing the whitespace trimming. Personally I would prefer to move all the canonicalization changes to `JsonView::canonicalize()` to be more modularized, and leave `JsonParseFunction::validate()` to do the simple validation as before.

Reply:
Yes, Agreed, moving the trim etc to the canonicalize().